### DISCUSSION TITLE
Long pauses loading Peggle levels on M1 Mac

### DISCUSSION THREAD

**Discussion by sood5cg** [The-Wineskin-Project/WineskinServer#331] [Q&A]
I've just moved to an M1 MBA running Monterey from an Intel Macbook running Mojave, so I'm afraid there are a lot of variables in play.  I installed the Peggle WoW edition on my Macbook using Wineskin and it worked fine.  I can do the same on my new MBA and it works fine... except that there's a roughly 15-second pause before the start of each level.  Once a level starts, it runs perfectly.  I've duplicated this same behavior on the Peggle Nights trial version.  On the Macbook, there's no discernible wait between clicking on a level and when that level starts being rendered.  On the new system, the music stops and nothing discernible happens for ~15 seconds before the level gets drawn.

I'd love suggestions on how to debug this.  I've done a test run and looked through the log, but I don't know what I'm looking for.  Any pointers?  Thanks in advance.

**Comment by Gcenx** [comment]
I believe the issue is due to Apple Silicon running OpenGL on-top of the Metal API.

Many Pokémon fan games also have a similar issue and I’ve found no way to resolve this.

**Comment by Kazbah** [comment]
I am having the same behavior in the ShadowBane client whcih is an old OpenGL game
